### ISSUE TITLE
[Feature]: Export a Copy/Paste config for direct device import.

### ISSUE THREAD

**Issue by cyberndj** [netbox-community/netbox-acls#55]
### NetBox version

v3.2.7

### Feature type

Add a function

### Proposed functionality

Add a export function to create a copy/paste ACL for adding to device.

### Use case

A link or button would generate a text file or webpage that has a premade configuration you could copy and paste into the device. 

### External dependencies

I would assume a platform specific ACL parser (ie for Cisco IOS, Juniper, Arista, etc...)

**Comment by ryanmerolle** [comment]
Not a horrible idea. I personally use NetBox as a source of truth that gets feed to templates.  I have not generate much of any templates in NetBox itself.  I think this would be more of a [export templates](https://docs.netbox.dev/en/stable/customization/export-templates/).  I would not be good at keeping up with the various vendor templates.

I could verify and test that export templates work with the model though.  I could provide examples as well.

Let me know if this is interesting.

**Comment by cyberndj** [comment]
Interesting to me, yes! I would be interested to get others input as well.
I would also keep it as a low priority compared to the other requests out there.

**Comment by cyberndj** [comment]
Saw this gem in base....I would suggest waiting on this until the base gets the fix. May be easier to implement too.

https://github.com/netbox-community/netbox/issues/11559

**Comment by abhi1693** [comment]
Since we utilize generic relations in the models, it should now be possible to render configuration natively. I have not yet tested this however I also dont see any issues why it would not work.

**Comment by pheus** [comment]
Thanks for raising this, and sorry this sat for a while.

NetBox already provides native export template support, which covers rendering text for copy/paste or download. A built-in device-specific ACL export here would also require vendor-specific rendering logic, which is outside the scope of this plugin because we want to keep it vendor-neutral.

For that reason, I’m going to close this as not planned. For vendor-specific config generation, export templates or external automation are a better fit.
